src/components/Filters/index.js

```diff
@@ -0,0 +1,327 @@
+      if (value !== this.state.searchString) {
+        this.setState({ searchString: value })
+      }
+    } else {
+      this.setState({ searchString: '' })
+    }
+  }
+
+  generateFreshEdit () {
+    if (!this.props.schema) {
+      return {}
+    }
+    const inputModels = sieve.makeFilterInputs(this.props.schema)
+
+    const edit = {
+      name: Object.keys(inputModels).shift(),
+      value: ''
+    }
+
+    edit.operator = inputModels[edit.name].availableOperators[0]
+
+    return edit
+  }
+
+  addFilterRule (rule) {
+    const { rules } = this.props
+    rules.push(rule)
+    this.props.setRules(rules)
+  }
+
+  editFilterRule (rule) {
+    const { rules } = this.props
+    const updatedRules = rules.map(r => (r.id === rule ? rule : r))
+
+    this.props.setRules(updatedRules)
+  }
+
+  addRule (rule) {
+    const inputModels = sieve.makeFilterInputs(this.props.schema)
+
+    if (!rule) {
+      rule = cloneDeep(this.state.edit)
+    }
+    const baseRule = inputModels[rule.name]
+    const newRule = assign(cloneDeep(baseRule), rule)
+
+    if (newRule.id) {
+      this.editFilterRule(newRule)
+    } else {
+      newRule.id = utils.randomString()
+      this.addFilterRule(newRule)
+    }
+    this.setState({
+      showModal: false,
+      edit: this.generateFreshEdit()
+    })
+  }
+
+  updateSimpleSearch (val) {
+    this.setState({ searchString: val })
+    const { rules } = this.props
+    const existingRule = find(rules, { name: sieve.SIMPLE_SEARCH_NAME })
+    if (existingRule) {
+      existingRule.value = val
+      this.editFilterRule(existingRule)
+    } else {
+      this.addFilterRule({
+        name: sieve.SIMPLE_SEARCH_NAME,
+        value: val,
+        id: utils.randomString()
+      })
+    }
+  }
+
+  toggleModal () {
+    this.setState({ showModal: !this.state.showModal })
+  }
+
+  showEditModal (rule) {
+    this.setState({
+      showModal: true,
+      edit: rule
+    })
+  }
+
+  removeRule (rule) {
+    if (rule.name === sieve.SIMPLE_SEARCH_NAME) {
+      this.setState({ searchString: '' })
+    }
+
+    const { rules } = this.props
+
+    const updatedRules = rules.filter(r => r.id !== rule.id)
+    this.props.setRules(updatedRules)
+  }
+
+  handleEditChange (value, attribute) {
+    const update = this.state.edit
+    const inputModels = sieve.makeFilterInputs(this.props.schema)
+
+    if (attribute === 'name' && update.name !== value) {
+      update.name = value
+      update.operator = inputModels[value].availableOperators[0]
+      update.value = ''
+    } else {
+      update[attribute] = value
+    }
+
+    this.setState({ edit: update })
+  }
+
+  saveView (name, scope) {
+    const { rules } = this.props
+    let { views } = this.props
+
+    const newView = {
+      name,
+      rules,
+      id: utils.randomString(),
+      scope
+    }
+
+    if (!views) {
+      views = {}
+    }
+
+    if (scope === 'global') {
+      if (!views.global) {
+        views.global = []
```

Comment:
How about `views.global = views.global || []` to make it a bit shallower?

Reply:
I think the current if statement is actually clearer in intention